### PR TITLE
Jugveer-Sandher/power_manager

### PR DESCRIPTION
**Pull request by Jugveer-Sandher**
### Summary
Created a PowerManger file for the PDM that can change the state of the efuses to shutdown or drive.

### Testing Done
I did some testing checking if the channel is open or closed based on the state passed into the PowerManager_SetState() func. Also checked that the PowerManager_Init() func sets the efuses to closed.

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,5 +6,9 @@
 			],
 			"url": "/can_bus/json/schema_tx.json"
 		}
-	]
+	],
+	"files.associations": {
+		"app_powermanager.h": "c"
+	},
+	"cmake.configureOnOpen": false
```
Comment (gtaharaedmonds) [inline .vscode/settings.json:13]: Can you revert these changes?

File: firmware/thruna/DIM/src/app/states/app_driveState.c [hunk 2]
```diff
@@ -99,7 +99,7 @@ static void driveStateRunOnTick100Hz(struct StateMachine *const state_machine)
 
     const float min_cell_voltage = App_CanRx_BMS_CellVoltages_MinCellVoltage_Get();
 
-    if (missing_hb)
+    if (false)
```
Comment (gtaharaedmonds) [inline firmware/thruna/DIM/src/app/states/app_driveState.c:102]: revert this too

File: firmware/thruna/PDM/Inc/App/App_PowerManager.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Initalizes to the drive state (closed)
+ */
+void PowerManager_Init();
+
+/**
+ * Sets the Power Manager state
+ * @param state to open/close
+ */
+void PowerManager_SetState(PowerManagerState state);
```
Comment (gtaharaedmonds) [inline firmware/thruna/PDM/Inc/App/App_PowerManager.h:20]: Prefix app-level functions with `App_`

File: firmware/thruna/PDM/Src/Io/tempCodeRunnerFile.c [hunk 4]
```diff
@@ -0,0 +1 @@
+#include "Io_Hal.h"
```
Comment (gtaharaedmonds) [inline firmware/thruna/PDM/Src/Io/tempCodeRunnerFile.c:1]: Delete this file

File: firmware/thruna/PDM/Src/App/App_powerManager.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            [EFUSE_CHANNEL_AIR] = true,
+            [EFUSE_CHANNEL_LVPWR] = true,
+            [EFUSE_CHANNEL_EMETER] = true,
+            [EFUSE_CHANNEL_AUX] = true,
+            [EFUSE_CHANNEL_DRS] = true,
+            [EFUSE_CHANNEL_FAN] = true,
+            [EFUSE_CHANNEL_DI_LHS] = true,
+            [EFUSE_CHANNEL_DI_RHS] = true,
+        }
+    }
```
Comment (gtaharaedmonds) [inline firmware/thruna/PDM/Src/App/App_powerManager.c:32]: Enabled channels from Joe:

- EMETER, AUX, and DRS should be always off
- AIR and LVPWR should be always on
- DI_LHS and DI_RHS should be on only in the drive state

File: firmware/thruna/PDM/Inc/App/App_PowerManager.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#include <stdbool.h>
+#include "Io_Efuse.h"
+
+typedef enum
+{
+    POWER_MANAGER_CONTRACTOR_OPEN,
+    POWER_MANAGER_CONTRACTOR_CLOSED,
```
Comment (gtaharaedmonds) [inline firmware/thruna/PDM/Inc/App/App_PowerManager.h:7]: Maybe rename `POWER_MANAGER_CONTRACTOR_OPEN` -> `POWER_MANAGER_STATE_SHUTDOWN`, and -> `POWER_MANAGER_CONTRACTOR_CLOSED` -> `POWER_MANAGER_STATE_DRIVE`. This is because there will be a few seconds before we close contactors where we'll enable high power to turn on the inverters.

File: firmware/thruna/PDM/Src/App/App_powerManager.c [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+void PowerManager_Init()
```
Comment (gtaharaedmonds) [inline firmware/thruna/PDM/Src/App/App_powerManager.c:35]: Make sure you call this init function somewhere in `main.c` for it to actually get run.

**Review by gtaharaedmonds — CHANGES_REQUESTED** [review]
Nice, looks great!

**Review by lehroatic — COMMENTED** [review]
Looks good, do address Gus's comments.

File: firmware/thruna/PDM/Test/Src/Test_PowerManager.cpp
Comment (gtaharaedmonds) [inline firmware/thruna/PDM/Test/Src/Test_PowerManager.cpp:1]: I think you can delete this file. In the tests, all we do is test that the fake function returns the value we set it to (true or false). So we're basically testing the fake function library itself, rather than our app-level code. 

There's currently no app code here to test - let's delete this file now and write some tests after you've written some more stuff in future tickets.

But I do like that you're thinking about tests :)

**Review by gtaharaedmonds — CHANGES_REQUESTED** [review]

File: firmware/thruna/PDM/Test/Src/Test_PowerManager.cpp
Comment (gtaharaedmonds) [reply firmware/thruna/PDM/Test/Src/Test_PowerManager.cpp:1]: Other than that this looks good to go in
